prime-router/src/main/kotlin/fhirengine/engine/FHIRReceiverFilter.kt

```diff
@@ -117,16 +118,56 @@ class FHIRReceiverFilter(
         override val errorCode: ErrorCode = ErrorCode.UNKNOWN
     }
 
-    data class FilterDetails(val filters: List<String>, val filterType: ReportStreamFilterType)
+    class AlternativeReceiverItemFilteredActionLogDetail(
+        val filter: String,
+        @JsonProperty
+        val filterType: String,
+        @JsonProperty
+        val receiverOrg: String,
+        @JsonProperty
+        val receiverName: String,
+        val index: Int = 1,
+    ) : ActionLogDetail {
+        override val scope: ActionLogScope = ActionLogScope.item
+
+        override val message: String = "Item was not routed to $receiverOrg.$receiverName because it did not pass the" +
+            " $filterType. Item failed on: $filter"
+
+        override val errorCode: ErrorCode = ErrorCode.UNKNOWN
+    }
+
+    interface FilterDetailsInterface {
+        fun theFilterType(): String
```

Comment:
On this project we do not follow the convention of "the" and "a" and "my". You can just use lower camel case without the prefix.